src/services/MessageAPI/index.js

```diff
@@ -0,0 +1,106 @@
+import axios from 'axios';
+import {BehaviorSubject} from 'rxjs';
+
+class MessageAPIService {
+  _sent_messages = new BehaviorSubject([]);
+  _received_messages = new BehaviorSubject([]);
+
+  sent_messages$ = this._sent_messages.asObservable();
+  received_messages$ = this._received_messages.asObservable();
+
+  async sendMessage(
+    message,
+    fingerprint,
+    signature,
+    publicKey,
+    sender_id,
+    recipient_id,
+    message_encryption_indicator_id
+  ) {
+    let retval = undefined;
+    await axios
+      .post(
+        `http://localhost:1234/api/messages/`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          }
+        },
+        {
+          message: message,
+          public_key: publicKey,
+          signature: signature,
+          fingerprint: fingerprint,
+          sender: `http://localhost:1234/api/identities/${sender_id}/`,
+          recipient: `http://localhost:1234/api/identities/${recipient_id}/`,
+          message_encryption_indicator: `http://localhost:1234/api/message_encryption_indicators/${message_encryption_indicator_id}/`
+        }
+      )
+      .then(function (response) {
+        console.log(response.data.results);
+        retval = response.data.results;
+      })
+      .catch(function (error) {
+        console.error(error);
+        retval = [];
+      });
+    this.__addSentMessage(retval);
+  }
+
+  async checkMessages(recipientID) {
+    let results = await axios
+      .get(`http://localhost:1234/api/messages/?recipient=${recipientID}/`, {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      })
+      .then(function (response) {
+        console.log(response.data.results);
+        return response.data.results;
+      })
+      .catch(function (error) {
+        console.error(error);
+        return [];
+      });
+      this.__populateReceivedMessages(results);
+  }
+
+  async getSentMessages(senderID) {
+    let retval = undefined;
+    await axios
+      .get(`http://localhost:1234/api/messages/?sender=${senderID}/`, {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      })
+      .then(function (response) {
+        console.log(response.data.results);
+        retval = response.data.results;
+      })
+      .catch(function (error) {
+        console.error(error);
+        retval = [];
+      });
+    this.__populateSentMessages(retval);
+  }
```

Comment:
I'd also use the pattern I suggested in `checkMessages` here as well concerning returning within the promise back to the `retval` or `results` variable